mesonbuild/interpreter.py

```diff
@@ -3168,10 +3188,13 @@ def get_subproject_dep(self, display_name, dirname, varname, kwargs):
         except InvalidArguments:
             pass
 
-        if not isinstance(dep, DependencyHolder):
+        if not isinstance(dep, (ExecutableHolder, DependencyHolder)):
```

Comment:
You can't accept both. When doing a dependency() fallback it must accept only DependencyHolder, and when doing a find_program() fallback it must accept only ExecutableHolder (and ExternalProgramHolder too I guess, but it would be weird to fallback to find a system program...).

Reply:
But I agree it's better to add support for both code paths here than duplicating all the code.